### DISCUSSION TITLE
How is `RefHeightAboveSfc` obtained, and the requirements for forcing data

### DISCUSSION THREAD

**Discussion by ZhouYanQiao** [NCAR/noahmp#143] [Q&A] [answered]
Hello, I am trying to run Noah-MP in a completely offline mode, **which means I do not want to use HRLDAS.** 

So far, I have collected the basic forcing data: **temperature at 2m, pressure at 2m, specific humidity at 2m, northward wind speed at 10m, eastward wind speed at 10m, longwave radiation absorbed at the surface, and shortwave radiation absorbed at the surface.** With this forcing data, I have successfully run Noah-MP without HRLDAS. 

However, the results seem incorrect, particularly the sensible heat flux. Therefore, I started checking all the variables. I found that a parameter called `RefHeightAboveSfc` is unknown, and in the technical manual, it is defined as "Reference Height above surface zero plane (including vegetation)." This parameter is indicated to be obtained through a climate model, but I am not using one. 

By reviewing all the code, I learned from the file `GroundRoughnessPropertyMod.F90` that `RefHeightAboveSfc = DZ8W(I,1,J) * 0.5`. `DZ8W(I,1,J)` is defined as the thickness of the lowest atmosphere layer. From the file `module_NoahMP_hrldas_driver.F90`, I found that `NoahmpIO%DZ8W = 2 * NoahmpIO%ZLVL`, where `ZLVL` is defined as the forcing height (which in Noah is considered to be the wind speed height, i.e., 10.0m). 

Thus, can we conclude that `RefHeightAboveSfc = NoahmpIO%DZ8W * 0.5 = 2 * NoahmpIO%ZLVL * 0.5 = 10.0m`? However, I do not understand why `DZ8W` in `GroundRoughnessPropertyMod.F90` is an array, while `DZ8W` in `module_NoahMP_hrldas_driver.F90` is not an array.

In summary, my questions can be summarized as follows:
**1. How is `RefHeightAboveSfc` obtained, and is it equal to 10.0m? Or how is `DZ8W` obtained
2. The forcing height is not unique; for example, the measurement height for temperature forcing data is 2.0m, but it seems Noah-MP does not require me to input the measurement height for temperature forcing data. Does this mean that the temperature forcing data input for Noah-MP must be at 2.0m?**

![image](https://github.com/user-attachments/assets/ffcdc402-f020-45c2-8917-11a72a118cdc)
![image](https://github.com/user-attachments/assets/aa13e23b-90cf-4da6-adf6-0217438dc26e)

The specific requirements for forcing data are not mentioned in the technical manual, likely because the default users obtain climate forcing data using HRLDAS. For example, the forcing input for temperature is referred to as "the surface air temperature" in the manual, but I find it difficult to understand how "surface" is defined in this context.

From the information I found about running Noah-MP with ERA5 data, it seems that the temperature forcing provided by ERA5 is the temperature at 2m. Does this mean that "surface" refers to a plane located 2m above the ground?

As a student with a background in computer science, I not have the expertise in this area, so I hope to receive a prompt response to clarify these questions.

![image](https://github.com/user-attachments/assets/412dc054-544b-475c-8f30-b4cd8fb2b20b)

**Comment by tslin2** [comment]
Hi,
1) RefHeightAboveSfc is the height that the atmospheric forcing provided to the land surface model. 
For the coupled atmospheric model, it is the midpoint of the first vertical of the atmospheric layer, which is the 0.5*DZ8W. DZ8W is the thickness of vertical atmospheric layers used in the coupled atmospheric model, such as WRF. If running in an offline land surface model like HRLDAS, forcing height is determined by the variable ZLVL in the namelist.

2) 2m temperature is the diagnostic output variable in the Noah-MP model. All the atmospheric forcing (including surface temperature) is input to Noah-MP at a specified height (typically 10 meters), and then Noah-MP calculates the 2-m temperature for vegetated and bare portions of each grid based on land surface energy balance. Finally, the gridmean 2-m temperature is a weighted average of the 2-m temperature for vegetated and bare portions of each grid. Thus, the resulting 2-m temperature will not be exactly the same as the input surface temperature forcing. (https://egusphere.copernicus.org/preprints/2023/egusphere-2023-675/egusphere-2023-675-AR1.pdf)

You can change the ZLVL in the namelist, If you know the specific height for these atmospheric variables, like measurement height from the flux tower or reference heights of the variables from the atmospheric models, like ERA5 reanalysis from the first atmospheric layer, you could provide the specific heights for each variable in the offline land model.

**Comment by ZhouYanQiao** [comment]
Hi!
Thank you very much for your prompt reply. Your reply has solved most of my doubts, but there are still the following points that I am looking forward to you answering again:
&nbsp;1. you mentioned that all forcing data are assumed to be at a specific height (commonly 10m). Does this mean that I input the temperature at 2m as forcing data, and then Noah-MP will base on this 2m temperature to calculate the temperature at 10m and use it as the actual climate forcing data? However, I didn't find anything about this in the Noah-MP code.
2. you mentioned that I can set a specific height for each forcing variable if I run Noah in offline mode. However I didn't find where to set these values in Noah's code. For example, I need to set the specific height for the temperature variable to 2.0m and the wind speed variable to 10.0m.
3. does ZLVL refer to the forced height of the wind speed?



thank you again!



---Original---
From: "Tzu-Shun ***@***.***&gt;
Date: Thu, Sep 19, 2024 23:48 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NCAR/noahmp] How is `RefHeightAboveSfc` obtained, and therequirements for forcing data (Discussion #143)




 
Hi,
  
 
RefHeightAboveSfc is the height that the atmospheric forcing provided to the land surface model.
 For the coupled atmospheric model, it is the midpoint of the first vertical of the atmospheric layer, which is the 0.5*DZ8W. DZ8W is the thickness of vertical atmospheric layers used in the coupled atmospheric model, such as WRF. If running in an offline land surface model like HRLDAS, forcing height is determined by the variable ZLVL in the namelist.
 
 
 
2m temperature is the diagnostic output variable in the Noah-MP model. All the atmospheric forcing (including surface temperature) is input to Noah-MP at a specified height (typically 10 meters), and then Noah-MP calculates the 2-m temperature for vegetated and bare portions of each grid based on land surface energy balance. Finally, the gridmean 2-m temperature is a weighted average of the 2-m temperature for vegetated and bare portions of each grid. Thus, the resulting 2-m temperature will not be exactly the same as the input surface temperature forcing. (https://egusphere.copernicus.org/preprints/2023/egusphere-2023-675/egusphere-2023-675-AR1.pdf)
 
  
If you know the specific height for these atmospheric variables, like measurement height from the flux tower or heights from the atmospheric models, like ERA5 reanalysis from the first atmospheric layer instead of 2m variables, you could provide the specific heights for each variable in the offline land model.

**Reply by tslin2** [reply]
Note that RefHeightAboveSfc is defined as reference height [m] above surface zero plane (including vegetation)

1. Current Noah-MP does not diagnose 10m temperature; had to give the forcing reference height in the namelist, ZLVL (default is 10m)
2. The reference height is currently setting the same for all forcing variables.
3. ZLVL is the RefHeightAboveSfc for forcing variables.

**Comment by ZhouYanQiao** [comment]
Thanks again! As you said, the specific heights of all forcing variables in noah are consistent to 10.0m, does this mean that I am wrong to use the temperature data at 2m as the temperature forcing data? However, usually temperature datasets at 10.0 m are not common. The same problem applies to using pressure data at 2m as barometric forcing data.
These are my last questions.
I apologize for bothering you!




---Original---
From: "Tzu-Shun ***@***.***&gt;
Date: Fri, Sep 20, 2024 02:13 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [NCAR/noahmp] How is `RefHeightAboveSfc` obtained, and therequirements for forcing data (Discussion #143)




 
Note that RefHeightAboveSfc is defined as reference height [m] above surface zero plane (including vegetation)
  
Current Noah-MP does not diagnose 10m temperature; had to give the forcing reference height in the namelist, ZLVL (default is 10m)
 
The reference height is currently setting the same for all forcing variables.
 
ZLVL is the RefHeightAboveSfc for forcing variables.

**Reply by tslin2** [reply, marked as answer]
One could do is to make the code changes to consider different heights for different forcing variables like temperature, wind, and specific humidity. My hypothesis is that the temperature differences are around 9.8 K/1000 m, so a temperature change of 8 meters is around 0.0784 K. The temperature changes likely has smaller effect compared to the choice of options for surface layer drag coeff (M-O or original Noah (Chen97))   
Another way is to adjust the temperature values to the specific height based on 9.8 K/1000 m rate

**Reply by ZhouYanQiao** [reply]
Thank you so much, you answered all my questions.
I now understand that although Noah in principle needs to enter the forcing variable at 10m, since the difference between the temperature at 2m and 10m is not very large, it is possible to enter the temperature at 2m directly
